### DISCUSSION TITLE
Database Support for forms and various registrations

### DISCUSSION THREAD

**Discussion by varshith257** [Anishkagupta04/RAPIDOC-HEALTHCARE-WEBSITE-#455] [Ideas]
PostgreSQL is one I prefer as I worked with three of these dbs. But I can't say we use this only for our project as every usecase has other preferences over dbs in projects. Submit a report of your findings, why we prefer PostgreSQL for our project, and why not MongoDB and MySQL. Pros and cons of them and what you are suggesting to use finally based on your research?

**Comment by varshith257** [comment]
@Pruthaa17 Here's a discussion for this. You can go ahead here as per https://github.com/Anishkagupta04/RAPIDOC-HEALTHCARE-WEBSITE-/pull/449#issuecomment-2152620158 

Once we finalise this, we can create issue from discussion and will assign you

**Reply by Pruthaa17** [reply]
As per my findings of using database , both MongoDB and PostgreSQL has there own pros  like : 
MongoDB having pros like flexible , easy to query , highly scalable  and good at handling large amounts of data and traffic .
PostgreSQL having table-oriented data model, reliable, secure and good at storing and managing complex data 

But the fact that PostgreSQL benefits from a strong open-source community with extensive documentation, support forums and longer history often translates to a broader base of community knowledge and third-party integrations. So Considering this I'm convinced for preferring PostgreSQL over rest .

**Reply by varshith257** [reply]
@Pruthaa17 Yes, one of the benefits of PostgreSQL is an open-source solution as well as supports complex transactions and also supports both relational and non relational DBs.

**Comment by varshith257** [comment]
@Pruthaa17 I think your next step could be to find out the tables and schema needed for our project. They can make it easier to implement them without any confusion or further modifications. For this, you can use https://drawsql.app/    or https://www.eraser.io/use-case/data-model  

For reference the following I did while I was preparing for the GSOC proposal:

![image](https://github.com/Anishkagupta04/RAPIDOC-HEALTHCARE-WEBSITE-/assets/117595548/798669a3-6ae4-4674-ab39-627250c8d40b)






> [!TIP]
> Specify the 1-1 and 1- many relationships between tables too

**Comment by Pruthaa17** [comment]
https://drawsql.app/teams/rapidoc/diagrams/db101 , Can you please review it and suggest some changes ? @varshith257

**Reply by varshith257** [reply]
@Pruthaa17 Will review it soon. Thanks for the work
